packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+    // outputs decrypted hex
+    // console.log(decipher.output.toHex());
+    return decipher.output.data;
+  }
+
+  /**
+   * @param {object} account - The account to be encrypted. Only the private key will be encrypted.
+   * @param {string} password - The password string used to encrypt the private key
+   * If on stellar, return the encrypted private key;
+   * If on ethereum, return the keystore V3 JSON.
+   * Plus all the auxiliary fields
+   * @returns the encrypted account
+   */
+  encrypt(account, password) {
+    switch (account.network) {
+      case 'stellar': {
+        // AES key and IV sizes
+        const keySize = 24;
+        const ivSize = 8;
+        // get derived bytes
+        const salt = forge.random.getBytesSync(8);
+        const derivedBytes = forge.pbe.opensslDeriveBytes(
+          password, salt, keySize + ivSize
+        ); /* , md */
+        const buffer = forge.util.createBuffer(derivedBytes);
+        const key = buffer.getBytes(keySize);
+        const iv = buffer.getBytes(ivSize);
+        const cipher = forge.cipher.createCipher('AES-CBC', key);
+        cipher.start({ iv });
+        cipher.update(forge.util.createBuffer(account.getPrivateKey(), 'binary'));
+        cipher.finish();
+        const output = forge.util.createBuffer();
+        // if using a salt, prepend this to the output:
+        if (salt !== null) {
+          output.putBytes('Salted__'); // (add to match openssl tool output)
+          output.putBytes(salt);
+        }
+        output.putBuffer(cipher.output);
+        return { original: output, network: this.network, balance: account.balance };
+      }
+
+      case 'ethereum':
+        return {
+          original: web3.eth.accounts.encrypt(account.getPrivateKey(), password),
+          network: this.network,
+          balance: account.balance
+        };
+        // return this.getOriginalAccount().encrypt(this.getPrivateKey(), password)
+      default:
+        console.log('The network is not correctly set');
+        return null;
+    }
+  }
+
+  /**
+   * @param {object} cipher - The encrypted object
+   * @param {string} password - The password that is used to encrypt the account
+   * Return an account, reconstructed from the cipher
+   * The fields remain the same, but methods are not.
+   * However, these methods will not be used, i.e. will use web3/stellar libraries.
+   * @returns {Account} the decrypted account
+   */
+  decrypt(cipher, password) {
+    switch (this.network) {
+      case 'stellar': {
+        // parse salt from input
+        const input = forge.util.createBuffer(cipher.original, 'binary');
+        // skip "Salted__" (if known to be present)
+        input.getBytes('Salted__'.length);
+        // read 8-byte salt
+        const salt = input.getBytes(8);
+        // AES key and IV sizes
+        const keySize = 24;
+        const ivSize = 8;
+        const derivedBytes = forge.pbe.opensslDeriveBytes(password, salt, keySize + ivSize);
```

Comment:
Is there a reason why openssl key derivation is used here? Why the difference in implementation with `encryptSeed`?